homeassistant/components/renson/__init__.py

```diff
@@ -0,0 +1,82 @@
+"""The Renson integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+import async_timeout
+from renson_endura_delta.renson import RensonVentilation
+from requests import RequestException
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = [
+    Platform.SENSOR,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Renson from a config entry."""
+
+    api = RensonVentilation(entry.data[CONF_HOST])
+    coordinator = RensonCoordinator("Renson", hass, api)
+
+    if not await hass.async_add_executor_job(api.connect):
+        raise ConfigEntryNotReady("Cannot connect to Renson device")
+
+    await coordinator.async_config_entry_first_refresh()
+
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = {
```

Comment:
Side note for the future: It's better to use a dataclass when storing multiple things in hass.data.

https://docs.python.org/3/library/dataclasses.html